frontend/src/actions/openstackActions.js

```diff
@@ -68,3 +112,150 @@ export const setOSOSortDir = (direction) => ({
+      getSelectedFilter(selectedCategory, selectedOption, "oso", isFromMetrics)
+    );
+    dispatch({
+      type: TYPES.SET_SELECTED_OSO_FILTERS,
+      payload: selectedFilters,
+    });
+  };
+export const setOSOAppliedFilters = (navigate) => (dispatch, getState) => {
+  const { start_date, end_date, selectedFilters } = getState().oso;
+  const appliedFilterArr = selectedFilters.filter((i) => i.value.length > 0);
+
+  const appliedFilters = {};
+  appliedFilterArr.forEach((item) => {
+    appliedFilters[item["name"]] = item.value;
+  });
+
+  dispatch({
+    type: TYPES.SET_OSO_APPLIED_FILTERS,
+    payload: appliedFilters,
+  });
+  appendQueryString({ ...appliedFilters, start_date, end_date }, navigate);
+  dispatch(applyFilters());
+};
+
+export const removeOSOAppliedFilters =
+  (filterKey, filterValue, navigate) => (dispatch, getState) => {
+    const appliedFilters = dispatch(
+      deleteAppliedFilters(filterKey, filterValue, "oso")
+    );
+    const { start_date, end_date } = getState().oso;
+    dispatch({
+      type: TYPES.SET_OSO_APPLIED_FILTERS,
+      payload: appliedFilters,
+    });
+    appendQueryString({ ...appliedFilters, start_date, end_date }, navigate);
+    dispatch(applyFilters());
+  };
+
+export const applyFilters = () => (dispatch) => {
+  dispatch(setOSOoffset(INITAL_OFFSET));
+  dispatch(setOSOPage(START_PAGE));
+  dispatch(fetchOSOJobs());
+  dispatch(buildFilterData());
+  dispatch(tableReCalcValues());
+};
+
+export const setOSODateFilter =
+  (start_date, end_date, navigate) => (dispatch, getState) => {
+    const appliedFilters = getState().oso.appliedFilters;
+
+    dispatch({
+      type: TYPES.SET_OSO_DATE_FILTER,
+      payload: {
+        start_date,
+        end_date,
+      },
+    });
+
+    appendQueryString({ ...appliedFilters, start_date, end_date }, navigate);
+  };
+
+export const applyOSODateFilter =
+  (start_date, end_date, navigate) => (dispatch) => {
+    dispatch(setOSOoffset(INITAL_OFFSET));
+    dispatch(setOSOPage(START_PAGE));
+    dispatch(setOSODateFilter(start_date, end_date, navigate));
+    dispatch(fetchOSOJobs());
+    dispatch(buildFilterData());
+  };
+
+export const buildFilterData = () => async (dispatch, getState) => {
+  try {
+    const { tableFilters, categoryFilterValue } = getState().oso;
+
+    const params = dispatch(getRequestParams("oso"));
+
+    const response = await API.get(API_ROUTES.OSO_FILTERS_API_V1, { params });
+    if (response.status === 200 && response?.data?.filterData?.length > 0) {
+      dispatch(getOSOSummary(response.data.summary));
+      dispatch({
+        type: TYPES.SET_OSO_FILTER_DATA,
+        payload: response.data.filterData,
+      });
+      const activeFilter = categoryFilterValue || tableFilters[0].name;
+      dispatch(setOSOCatFilters(activeFilter));
+    }
+  } catch (error) {
+    console.error(
+      `ERROR (${error?.response?.status}): ${JSON.stringify(
+        error?.response?.data
+      )}`
+    );
```

Comment:
These are internal errors we wouldn't generally expect the user to be able to do anything about; but we're essentially silently ignoring the error here by "hiding it" in the console output (which a typical user won't see). Shouldn't these be a Toast? (I know, this issue isn't unique to this change, and I won't hold up the PR, but it's an interesting "usability" issue.)